core/notification/notification.go

```diff
@@ -0,0 +1,59 @@
+package notification
+
+import (
+	"context"
+	"time"
+
+	"github.com/odpf/siren/pkg/errors"
+)
+
+//go:generate mockery --name=Notifier -r --case underscore --with-expecter --structname Notifier --filename notifier.go --output=./mocks
+type Notifier interface {
+	ValidateConfigMap(notificationConfigMap map[string]interface{}) error
```

Comment:
```suggestion
	ValidateConfig(notificationConfigMap map[string]interface{}) error
```

Reply:
I deliberately called this `ConfigMap` instead of `Config` only.